src/App.tsx

```diff
@@ -14,11 +14,58 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: 'Jam',
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <>
+        <div className="App">
+          <h1>
+            {selectedGoods
+              ? `${selectedGoods} is selected`
+              : 'No goods selected'}
+            <button
+              className={selectedGoods ? 'enable' : 'disable'}
+              type="button"
+              onClick={() => {
+                this.setState({
+                  selectedGoods: '',
+                });
+              }}
+            >
+              X
+            </button>
+          </h1>
+        </div>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li key={good} className={selectedGoods === good ? 'selected' : 'notSelected'}>
```

Comment:
next time use classnames lib for conditional classes
https://www.npmjs.com/package/classnames